Test failure(s): FirmwareUpdateServiceOSGiTest: correct firmware status

https://travis-ci.org/eclipse/smarthome/builds/237073476

```
assert that correct firmware status is provided(org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateServiceOSGiTest)  Time elapsed: 10.16 sec  <<< FAILURE!
java.lang.AssertionError: 
Expected: is <3>
     but: was <2>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert.assertThat(Assert.java:832)
	at org.junit.Assert$assertThat.callStatic(Unknown Source)
	at org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateServiceOSGiTest$_setup_closure5.doCall(FirmwareUpdateServiceOSGiTest.groovy:246)
	at org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateServiceOSGiTest$_setup_closure5.doCall(FirmwareUpdateServiceOSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor91.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:278)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:220)
	at sun.reflect.GeneratedMethodAccessor51.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrap.invoke(PogoMetaMethodSite.java:187)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:153)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:191)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor56.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateServiceOSGiTest.setup(FirmwareUpdateServiceOSGiTest.groovy:245)
```

[log.txt](https://github.com/eclipse/smarthome/files/1035384/log.txt)

```text
assert that firmware status is available instead of executable if update is not executable(org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateServiceOSGiTest)  Time elapsed: 0.08 sec  <<< FAILURE!
java.lang.AssertionError: 
Expected: is not null
     but: was null
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert.assertThat(Assert.java:832)
	at org.junit.Assert$assertThat.callStatic(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:53)
	at org.junit.Assert$assertThat.callStatic(Unknown Source)
	at org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateServiceOSGiTest.setup(FirmwareUpdateServiceOSGiTest.groovy:210)
assert that firmware status is available instead of executable if update is not executable(org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateServiceOSGiTest)  Time elapsed: 0.081 sec  <<< ERROR!
java.lang.NullPointerException: Cannot invoke method getAll() on null object
	at org.codehaus.groovy.runtime.NullObject.invokeMethod(NullObject.java:88)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:45)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.NullCallSite.call(NullCallSite.java:32)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.eclipse.smarthome.core.common.registry.Provider$getAll.call(Unknown Source)
	at org.eclipse.smarthome.core.thing.firmware.FirmwareUpdateServiceOSGiTest.teardown(FirmwareUpdateServiceOSGiTest.groovy:253)
```

--- a/bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateServiceOSGiTest.groovy
+++ b/bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateServiceOSGiTest.groovy
@@ -206,8 +206,11 @@ final class FirmwareUpdateServiceOSGiTest extends OSGiTest {
         setDefaultLocale(Locale.ENGLISH)
 
         registerVolatileStorageService()
-        managedThingProvider = getService ManagedThingProvider
-        assertThat managedThingProvider, is(notNullValue())
+        waitForAssert {
+            managedThingProvider = getService ManagedThingProvider
+            assertThat managedThingProvider, is(notNullValue())
+        }
+        
         thingRegistry = getService(ThingRegistry)
         assertThat thingRegistry, is(notNullValue())
 
@@ -249,11 +252,11 @@ final class FirmwareUpdateServiceOSGiTest extends OSGiTest {
 
     @After
     void teardown() {
-        setDefaultLocale(_defaultLocale)
         managedThingProvider.getAll().each {
             managedThingProvider.remove(it.getUID())
         }
         unregisterMocks()
+        setDefaultLocale(_defaultLocale)
     }
 
     @Test

